wiring/inc/spark_wiring_system.h

```
@@ -846,6 +875,37 @@ class SystemClass {
 
 #endif // HAL_PLATFORM_POWER_MANAGEMENT
 
+    int setControlRequestFilter(SystemControlRequestAclAction defaultAction, Vector<SystemControlRequestAclClass> filters = {}) {
+		// Add requests and their actions to the control filter
+        system_control_acl default_action = (system_control_acl)static_cast<typename std::underlying_type<SystemControlRequestAclAction>::type>(defaultAction);
+        system_control_filter* first = nullptr;
+        system_control_filter* current = nullptr;
+
+        for(int i=0; i<filters.size(); i++) {
+            if (first == nullptr) {
+                first = (system_control_filter*)malloc(sizeof(system_control_filter));
+                if (!first) {
+                    return SYSTEM_ERROR_NO_MEMORY;
+                }
+                first->req = filters[i].type();
+                first->action = (system_control_acl)static_cast<typename std::underlying_type<SystemControlRequestAclAction>::type>(filters[i].action());
+                LOG(TRACE, "TP1 - action: %d", first->action);
+                first->next = nullptr;
+                current = first;
+            } else {
+                system_control_filter* temp = (system_control_filter*)malloc(sizeof(system_control_filter));
```

allocation failure is not handled.